src/main/java/gregtech/api/metatileentity/TickableTileEntityBase.java

```diff
@@ -2,14 +2,33 @@
 
 import net.minecraft.util.ITickable;
 
+import java.util.Random;
+
 public abstract class TickableTileEntityBase extends SyncedTileEntityBase implements ITickable {
 
     private long timer = 0L;
 
+    // Create an offset [0,20) to distribute ticks more evenly
+    private Random random = new Random();
```

Comment:
I don't feel like we need to have Random in every `TickableTileEntityBase` when we use it only once. How about getting random int from some utility class. And maybe even XSTR when we are at it.

Reply:
Moved this to `GTUtility` as a private field, and added a public method `getRandomIntXSTR(int bound)`